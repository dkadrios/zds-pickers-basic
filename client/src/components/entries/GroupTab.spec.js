import flow from 'lodash/fp/flow'
import set from 'lodash/fp/set'
import WrappedGroupTab, { boxTarget } from './GroupTab'

const GroupTab = WrappedGroupTab.DecoratedComponent

const connectDropTarget = jest.fn()

const props = {
  active: true,
  connectDropTarget,
  canDrop: true,
  isOver: false,
  groupId: 1,
  selected: false,
  controlLabels: [],
}

describe('GroupTab tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(GroupTab)(props).toMatchSnapshot()
  })

  it('renders correctly when over', () => {
    shallowExpect(GroupTab)(set('isOver', true)(props)).toMatchSnapshot()
  })

  it('renders correctly when not allowed to drop', () => {
    shallowExpect(GroupTab)(flow(
      set('isOver', true),
      set('canDrop', false),
    )(props)).toMatchSnapshot()
  })

  it('renders correctly when not active', () => {
    shallowExpect(GroupTab)(set('active', false)(props)).toMatchSnapshot()
  })

  describe('boxTarget tests', () => {
    const groupId = 3

    beforeEach(jest.clearAllMocks)

    it('drop should return info on the group', () => {
      expect(boxTarget.drop({ groupId })).toEqual({ name: 'TAB', groupId })
    })

    it('can drop on self', () => {
      const monitor = {
        getItem: () => ({ groupId: 0 }),
      }
      expect(boxTarget.canDrop({ groupId }, monitor)).toBe(true)
    })

    it('cannot drop on self', () => {
      const monitor = {
        getItem: () => ({ groupId }),
      }
      expect(boxTarget.canDrop({ groupId }, monitor)).toBe(false)
    })
  })
})
