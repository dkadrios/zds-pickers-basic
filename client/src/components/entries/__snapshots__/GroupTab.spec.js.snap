// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupTab tests renders correctly 1`] = `
<GroupTab
  active={true}
  canDrop={true}
  classes={
    Object {
      "activeLabel": "GroupTab-activeLabel-1",
      "controlsLabel": "GroupTab-controlsLabel-2",
      "dropTarget": "GroupTab-dropTarget-3",
      "dropTargetOver": "GroupTab-dropTargetOver-4",
    }
  }
  connectDropTarget={[MockFunction]}
  controlLabels={Array []}
  groupId={1}
  isOver={false}
  selected={false}
/>
`;

exports[`GroupTab tests renders correctly when not active 1`] = `
<GroupTab
  active={false}
  canDrop={true}
  classes={
    Object {
      "activeLabel": "GroupTab-activeLabel-1",
      "controlsLabel": "GroupTab-controlsLabel-2",
      "dropTarget": "GroupTab-dropTarget-3",
      "dropTargetOver": "GroupTab-dropTargetOver-4",
    }
  }
  connectDropTarget={[MockFunction]}
  controlLabels={Array []}
  groupId={1}
  isOver={false}
  selected={false}
/>
`;

exports[`GroupTab tests renders correctly when not allowed to drop 1`] = `
<GroupTab
  active={true}
  canDrop={false}
  classes={
    Object {
      "activeLabel": "GroupTab-activeLabel-1",
      "controlsLabel": "GroupTab-controlsLabel-2",
      "dropTarget": "GroupTab-dropTarget-3",
      "dropTargetOver": "GroupTab-dropTargetOver-4",
    }
  }
  connectDropTarget={[MockFunction]}
  controlLabels={Array []}
  groupId={1}
  isOver={true}
  selected={false}
/>
`;

exports[`GroupTab tests renders correctly when over 1`] = `
<GroupTab
  active={true}
  canDrop={true}
  classes={
    Object {
      "activeLabel": "GroupTab-activeLabel-1",
      "controlsLabel": "GroupTab-controlsLabel-2",
      "dropTarget": "GroupTab-dropTarget-3",
      "dropTargetOver": "GroupTab-dropTargetOver-4",
    }
  }
  connectDropTarget={[MockFunction]}
  controlLabels={Array []}
  groupId={1}
  isOver={true}
  selected={false}
/>
`;
